eng/pipelines/agent-pool-migration.yml

```diff
@@ -0,0 +1,134 @@
+# 5. Go to the PR link in 'Create pull request' step. Review and merge the PR. Note, the pipeline intented to be aggressive, so take a closer look at the changes.
+#    The PR also has a link pointed back to the pipeline, so people will learn where it comes from and the purpose of the PR.
+parameters:
+- name: Repos
+  type: object
+  default:
+    # The exclude/include paths are valid for ubuntu agents. If use windows, please follow the format like "^.\\sdk\\" 
+    - RepoOwner: Azure
+      RepoName: azure-sdk-tools
+      ExcludePaths: ''
+      IncludePaths: ''
+      Branch: 'main'
+    - RepoOwner: Azure
+      RepoName: azure-sdk
+      ExcludePaths: "^./eng/common/*"
+      IncludePaths: ''
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-android
+      ExcludePaths: "^./sdk/|^./eng/common/|/tests/resources/session-records/"
+      IncludePaths: "^./sdk/.*-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-c
+      ExcludePaths: "^./sdk/|^./eng/common/"
+      IncludePaths: "^./sdk/.*-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-cpp
+      ExcludePaths: "^./sdk/|^./eng/common/"
+      IncludePaths: "^./sdk/.*-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-go
+      ExcludePaths: "^./sdk/|^./eng/common/"
+      IncludePaths: "^./sdk/.*-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-ios
+      ExcludePaths: "^./sdk/|^./eng/common/"
+      IncludePaths: "^./sdk/.*-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-java
+      ExcludePaths: "^./sdk/|^./eng/common/|/resources/session-records/"
+      IncludePaths: "^./sdk/.*-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-js
+      ExcludePaths: "^./sdk/|^./eng/common/|/recordings/node/"
+      IncludePaths: "^./sdk/.*-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-net
+      ExcludePaths: "^./sdk/|^./eng/common/|/tests/SessionRecords/"
+      IncludePaths: "^./sdk/.*-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-python
+      ExcludePaths: "^./sdk/|^./eng/common/|/tests/recordings/"
+      IncludePaths: "^./sdk/.*-matrix.*.json$|^./sdk/.*/.*yml$"
+# Default setting is an example of the map looks like:
+- name: MigrationMap
+  type: object
+  default: 
+    - MigrateFrom: azsdk-pool-mms-win-2022-general
+      MigrateTo: azsdk-pool-mms-win-2022-general
+    - MigrateFrom: windows-2022
+      MigrateTo: windows-2022
+    - MigrateFrom: windows2022
+      MigrateTo: windows2022
+    - MigrateFrom: MMS2019
+      MigrateTo: MMS2022
+- name: ShortForMigrateFrom
+  type: string
+  default: ''
+- name: ShortForMigrateTo
+  type: string
+  default: ''
+- name: RequestBy
```

Comment:
Lets change this to GithubAssigneeAlias so it is clear about what it is used for.